File: src/models/chat_session.py
from datetime import datetime

from pydantic import BaseModel


class ChatSession(BaseModel):
    id: str
    world_id: str
    type: str  # event | character
    title: str | None
    created_at: datetime
    participants: list[dict] | list[str] | None = None  # UUID array or legacy dict array
    participant_mode: str | None = None
    memories_enabled: bool = False
    version_id: str | None = None
    last_flushed_sequence: int = 0
    last_active_at: datetime | None = None
    element_injection_ids: list[str] | None = None
    constraints: str = ""


class ChatSessionListResponse(BaseModel):
    sessions: list[ChatSession]

File: frontend/src/types/chatSession.ts
export interface Participant {
  id: string
  name: string
}

export interface ChatSession {
  id: string
  world_id: string
  type: 'event' | 'character'
  title: string | null
  created_at: string
  last_active_at?: string | null
  participants: Participant[] | null
  participant_mode: 'auto' | 'edit' | 'include' | null
  element_injection_ids?: string[] | null
  constraints?: string
}

export interface ChatSessionListResponse {
  sessions: ChatSession[]
}
